crates/ruff_python_parser/tests/snapshots/invalid_syntax@decorator_missing_expression.py.snap

```diff
@@ -0,0 +1,189 @@
+---
+source: crates/ruff_python_parser/tests/fixtures.rs
+input_file: crates/ruff_python_parser/resources/inline/err/decorator_missing_expression.py
+---
+## AST
+
+```
+Module(
+    ModModule {
+        range: 0..51,
+        body: [
+            AnnAssign(
+                StmtAnnAssign {
+                    range: 5..15,
+                    target: Call(
+                        ExprCall {
+                            range: 5..10,
+                            func: Name(
+                                ExprName {
+                                    range: 5..8,
+                                    id: "foo",
+                                    ctx: Load,
+                                },
+                            ),
+                            arguments: Arguments {
+                                range: 8..10,
+                                args: [],
+                                keywords: [],
+                            },
+                        },
+                    ),
+                    annotation: EllipsisLiteral(
+                        ExprEllipsisLiteral {
+                            range: 12..15,
+                        },
+                    ),
+                    value: None,
+                    simple: false,
+                },
+            ),
+            FunctionDef(
```

Comment:
Huh, the AST here is a bit strange. Why is it not a decorator with a missing name?

Reply:
The example in this case is:

```py
@def foo(): ...
```

Here, the parser sees `def` which doesn't match any expression start and fallback to `parse_name` as it's a keyword. Then it expects a NEWLINE as each decorator node ends with it but doesn't find it. The parser exists the decorator parsing loop.

Following token is not the start of function or class as only they can have the decorators so fallback to parsing `foo(): ...` as a statement. This looks like an annotated assignment statement so it parses it as such.

Reply:
We could check if it's the start of an expression and only parse it if it is otherwise use an empty name node (not ideal, I know). Or rather I think I've a better solution.

Reply:
(I'll put it up as a separate PR.)